packages/cdktf/lib/validateEnvironment.ts

```diff
@@ -0,0 +1,17 @@
+import { Construct } from "constructs";
+
+// fails early if the environment contains e.g. invalid version combinations
+export function validateEnvironment() {
+  validateConstructsIsV10();
+}
+
+function validateConstructsIsV10() {
```

Comment:
Shouldn't this fail on installation already?

Reply:
For Golang yes, for npm it's a warning and for Java there's no error as far as I tested.